plugin/src/features/Compilation/index.tsx

```diff
@@ -36,45 +38,66 @@ export const Compilation = ({ setAccordian }: CompilationProps) => {
 
   const setContracts = useSetAtom(contractsAtom)
   const setSelectedContract = useSetAtom(selectedContractAtom)
+  const setDeployStatus = useSetAtom(deployStatusAtom)
 
   const { status, isCompiling } = useAtomValue(compilationAtom)
 
   const setCompileStatus = useSetAtom(compileStatusAtom)
   const setIsCompiling = useSetAtom(isCompilingAtom)
+  const [isCompilingProject, setIsCompilingProject] = useState(false)
   const setCompileErrorMessages = useSetAtom(compileErrorMessagesAtom)
+  const [isContractsFolderAvailable, setIsContractsFolderAvailable] = useState(true)
 
   const solidityVersion = useAtomValue(solidityVersionAtom)
 
-  const compilations = [
-    {
-      validation: isValidSolidity,
-      isLoading: isCompiling,
-      onClick: compile,
-      id: 1
-    }
-  ]
+  useEffect(() => {
+    remixClient.fileManager.readdir(`${currentWorkspacePath}/`).then((workspaceFiles: any) => {
+      if (!workspaceFiles || !workspaceFiles['contracts'] || workspaceFiles['contracts']['isDirectory'] === false) {
+        setIsContractsFolderAvailable(false)
+      } else {
+        setIsContractsFolderAvailable(true)
+      }
+    })
+  }, [currentWorkspacePath, remixClient])
 
-  async function compile(): Promise<void> {
-    setIsCompiling(true)
+  async function handleCompile({ type }: { type: 'PROJECT' | 'SINGLE_FILE' }): Promise<void> {
```

Comment:
Yeah I agree but I didn't went for changing the existing method too much in this PR. this will needs to be done subsequently as we continue with refactoring.

Reply:
> I didn't went for changing the existing method

There's good portion added and rewritten where it's already hard to read and maintain the code